enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,330 @@
+and integrate support into SecurityContextContraints (SCC).
+
+This feature relies on work already done in the kernel to support idmapped mounts, or a mechanism to allow filesystems to be user namespaces aware.
+This work was merged in RHEL 9.4.
+
+## Motivation
+
+Originally implemented in the linux kernel 3.8, user namespaces have long been a goal of the Kubernetes community.
+[KEP-127](www.github.com/kubernetes/enhancements/issues/127) is one of the oldest still-open KEPs today.
+Part of the push for user namespaces is it gets containers closer to the aspirational goal of a virtualized host: putting a process
+in a user namespace means it can have "privileges" inside the container, while being unprivileged on the host. Further, this
+divide between the container's namespace and the host's means an admin can allow users to gain access to privileges within the container,
+while being able to trust that the kernel doesn't grant them on the host. A consequence of this is users can, for instance, run podman
+within an Openshift pod without being in a privileged namespace.
+
+### User Stories
+
+* As an Openshift user, I would like to be able to run my container as root without needing to be trusted by the platform.
+* As a user of Openshift Devspaces, I would like to run podman within the Devspace.
+* As an Openshift admin, I would like to run untrusted users on a tighter security profile than the SCC restricted-v2.
+* As an Openshift admin, I would like to ensure pods that request user namespaces are confined to one.
+
+### Goals
+
+- Enable Openshift users to request a pod be put in a user namespace
+- Update SCC to take user namespaces into account when choosing the security profile of a container
+- Add support for users to run podman within an Openshift pod without being privileged.
+
+### Non-Goals
+
+- 
+
+## Proposal
+
+There are two pieces to this proposal:
+- Extend SCC to be aware of the hostUsers field:
+    - Add a new field HostUsers to SCC, which will be Allowed by default
+    - Add a new SCC to the default list: restricted-v3
+        - This SCC will be identical to restricted-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nonroot-v3
+        - This SCC will be identical to nonroot-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nested-container. It will have:
+        - SELinux context set to MustRunAs.Type: container_engine_t
+        - RunAsUserStrategy: RunAsAny
+        - RequiredDropCapabilities: None
+        - HostUsers: Disallowed
+        - And otherwise mirror the restricted-v2 profile
+- Add the features UserNamespacesPodSecurityStandards and ProcMountType to the list that qualify a cluster as TechPreviewNoUpgrade
+    - UserNamespacesSupport will be on by default, as the feature is beta in 1.30.
+
+### Workflow Description
+
+N/A
```

Comment:
Perhaps a workflow here would demonstrate how a user would migrate their existing workloads to the new policies? What do they need to set on the podSpec to disallow host users?

Reply:
Added one